dbms/src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -355,8 +355,12 @@ struct StorageDistributedDirectoryMonitor::Batch
             /// we must try to re-send exactly the same batches.
             /// So we save contents of the current batch into the current_batch_file_path file
             /// and truncate it afterwards if all went well.
-            WriteBufferFromFile out{parent.current_batch_file_path};
-            writeText(out);
+            {
+                WriteBufferFromFile out{parent.current_batch_file_path + ".tmp"};
+                writeText(out);
+                out.sync();
```

Comment:
This is irrelevant to proposed modifications.

Reply:
Let's remove it, because it may slow-down other fs operations due to introduction of barriers.

Reply:
>This is irrelevant to proposed modifications.

Agree, will be removed

> may slow-down other fs operations due to introduction of barriers.

I though that durability will not be superfluous, but I guess it is not that important.